File: docs/modules/examples/examples/clients/chat-room/nodejs/src/producer.ts
import { redpanda, topic } from "./config";

// tag::producer[]
const producer = redpanda.producer();

// Connect once and return a function that sends one JSON-encoded chat message.
export async function getConnection(user: string) {
  await producer.connect();
  return async (message: string) => {
    await producer.send({
      topic,
      messages: [{ value: JSON.stringify({ user, message }) }],
    });
  };
}
// end::producer[]

export async function disconnect() {
  await producer.disconnect();
}

File: docs/modules/examples/examples/clients/chat-room/go/producer.go
package main

import (
	"context"
	"encoding/json"

	"github.com/twmb/franz-go/pkg/kgo"
)

// tag::producer[]
type Producer struct {
	client *kgo.Client
	topic  string
}

func NewProducer(topic string) *Producer {
	client, err := kgo.NewClient(clientOpts()...)
	if err != nil {
		panic(err)
	}
	return &Producer{client: client, topic: topic}
}

// SendMessage produces one JSON-encoded chat message.
func (p *Producer) SendMessage(user, message string) {
	b, _ := json.Marshal(Message{User: user, Message: message})
	p.client.Produce(context.Background(), &kgo.Record{Topic: p.topic, Value: b}, nil)
}
// end::producer[]

func (p *Producer) Close() {
	p.client.Close()
}
